src/ManagedServices/ManagedServices/Commands/GetAzureRmManagedServicesDefinition.cs

```diff
@@ -0,0 +1,77 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.ManagedServices.Models;
+using Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Commands
+{
+    [Cmdlet(
+    VerbsCommon.Get,
+    Microsoft.Azure.Commands.ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ManagedServicesDefinition",
+    DefaultParameterSetName = DefaultParameterSet,
+    SupportsShouldProcess = true), OutputType(typeof(PSRegistrationDefinition))]
+    public class GetAzureRmManagedServicesDefinition : ManagedServicesCmdletBase
+    {
+        protected const string DefaultParameterSet = "Default";
+
+        [Parameter(Position = 0, ParameterSetName = DefaultParameterSet, HelpMessage = "The subscription scope.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = DefaultParameterSet, ValueFromPipeline = true, HelpMessage = "The registration definition idenfier.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("Id")]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var definitionId = this.ResourceId.IsGuid() ? this.ResourceId : this.ResourceId?.GetResourceName();
+            this.TryCatchExecuteCmdlet(() =>
+            {
+                ConfirmAction(MyInvocation.InvocationName,
+                    "Microsoft.ManagedServices/registrationDefinitions",
+                    () =>
+                    {
+                        var scope = this.Scope ?? this.GetDefaultScope();
+
+                        if (string.IsNullOrEmpty(definitionId))
+                        {
+                            var results = this.PSManagedServicesClient.ListRegistrationDefinitions(scope);
+                            this.WriteRegistrationDefinitionsList(results);
+                        }
+                        else
+                        {
+                            // validate definitionId.
+                            if (!definitionId.IsGuid())
+                            {
+                                throw new ApplicationException("RegistrationDefinitionId must be a valid GUID.");
+                            }
+
+                            var result = this.PSManagedServicesClient.GetRegistrationDefinition(
+                                scope: scope,
```

Comment:
Again, no, Scope cannot be required - either the subscriptionId comes from the resource id, or, if the user supplies a guid, then the current subscription shoudl eb used.

Reply:
Yup.